File: src/main/java/models/reservation.java
package models;

public class reservation {
    private int id_R;
    private int nombre_places;
    private String type_reservation;
    private int code_confirmation;
    private String remarque;
    private Evenement evenement;

    // Constructeur avec tous les attributs
    public reservation(int id_R, int nombre_places, String type_reservation, int code_confirmation, String remarque, Evenement evenement) {
        this.id_R = id_R;
        this.nombre_places = nombre_places;
        this.type_reservation = type_reservation;
        this.code_confirmation = code_confirmation;
        this.remarque = remarque;
        this.evenement = evenement;
    }

    // Constructeur sans id_R
    public reservation(int nombre_places, String type_reservation, int code_confirmation, String remarque, Evenement evenement) {
        this.id_R = 0;  // Valeur par défaut pour id_R
        this.nombre_places = nombre_places;
        this.type_reservation = type_reservation;
        this.code_confirmation = code_confirmation;
        this.remarque = remarque;
        this.evenement = evenement;
    }

    // Getters et setters
    public int getId_R() {
        return id_R;
    }

    public void setId_R(int id_R) {
        this.id_R = id_R;
    }

    public int getNombre_places() {
        return nombre_places;
    }

    public void setNombre_places(int nombre_places) {
        this.nombre_places = nombre_places;
    }

    public String getType_reservation() {
        return type_reservation;
    }

    public void setType_reservation(String type_reservation) {
        this.type_reservation = type_reservation;
    }

    public int getCode_confirmation() {
        return code_confirmation;
    }

    public void setCode_confirmation(int code_confirmation) {
        this.code_confirmation = code_confirmation;
    }

    public String getRemarque() {
        return remarque;
    }

    public void setRemarque(String remarque) {
        this.remarque = remarque;
    }

    public Evenement getEvenement() {
        return evenement;
    }

    public void setEvenement(Evenement evenement) {
        this.evenement = evenement;
    }
}

File: src/Entity/Reservation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use App\Repository\ReservationRepository;

#[ORM\Entity(repositoryClass: ReservationRepository::class)]
#[ORM\Table(name: 'reservation')]
class Reservation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id_R = null;

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $nombre_places = null;

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $type_reservation = null;

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $code_confirmation = null;

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $remarque = null;

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $nom = null;

    #[ORM\OneToMany(targetEntity: Paiement::class, mappedBy: 'reservation', cascade: ['persist', 'remove'])]
    private Collection $paiements;

    public function __construct()
    {
        $this->paiements = new ArrayCollection();
    }

    public function getId_R(): ?int
    {
        return $this->id_R;
    }

    public function setId_R(int $id_R): self
    {
        $this->id_R = $id_R;
        return $this;
    }

    public function getNombre_places(): ?int
    {
        return $this->nombre_places;
    }

    public function setNombre_places(int $nombre_places): self
    {
        $this->nombre_places = $nombre_places;
        return $this;
    }

    public function getType_reservation(): ?string
    {
        return $this->type_reservation;
    }

    public function setType_reservation(string $type_reservation): self
    {
        $this->type_reservation = $type_reservation;
        return $this;
    }

    public function getCode_confirmation(): ?int
    {
        return $this->code_confirmation;
    }

    public function setCode_confirmation(int $code_confirmation): self
    {
        $this->code_confirmation = $code_confirmation;
        return $this;
    }

    public function getRemarque(): ?string
    {
        return $this->remarque;
    }

    public function setRemarque(string $remarque): self
    {
        $this->remarque = $remarque;
        return $this;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;
        return $this;
    }

    /**
     * @return Collection<int, Paiement>
     */
    public function getPaiements(): Collection
    {
        return $this->paiements;
    }

    public function addPaiement(Paiement $paiement): self
    {
        if (!$this->paiements->contains($paiement)) {
            $this->paiements[] = $paiement;
            $paiement->setReservation($this); // Lier la réservation au paiement
        }
        return $this;
    }

    public function removePaiement(Paiement $paiement): self
    {
        if ($this->paiements->removeElement($paiement)) {
            // On dissocie la réservation du paiement
            if ($paiement->getReservation() === $this) {
                $paiement->setReservation(null);
            }
        }
        return $this;
    }
}
